JSQMessagesViewController/Controllers/JSQMessagesViewController.m

```diff
@@ -912,6 +934,16 @@ - (void)keyboardController:(JSQMessagesKeyboardController *)keyboardController k
 
     heightFromBottom = MAX(0.0, heightFromBottom);
 
+    // Hide the picker view if the keyboard/input view is dismissed via the
+    // pan gesture recognizer
+
+    if (heightFromBottom == 0.0 && self.isPickerViewVisible) {
+        self.isPickerViewVisible = NO;
+    }
+
+    // Show/hide the ad container view
+    self.isAdVisible = heightFromBottom == 0.0;
```

Comment:
Is this sufficient? One must also determine free/pro state for the member, in addition to tricky calculations which determine if a picker is showing and therefore hiding our ad view